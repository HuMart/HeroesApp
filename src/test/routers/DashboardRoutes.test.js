import React from "react";

import { mount } from "enzyme";
import { DashboardRoutes } from "../../routers/DashboardRoutes";
import { AuthContext } from "../../auth/authContext";
import { MemoryRouter } from "react-router-dom";



describe('test in <DashboardRoutes />', () => {
  
    test('should render correctli in marvel screen', () => {
        
        const contextValue = {
            user: {
                logged: true,
                name: 'User logged'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={ ['/'] }>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        )

        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.text-info').text().trim() ).toBe('User logged');
        expect( wrapper.find('h1').text().trim() ).toBe('Marvel Screen');


    });
    
    test('should render correctli in dc screen', () => {
        
        const contextValue = {
            user: {
                logged: true,
                name: 'User logged'
            }
        }

        const wrapper = mount(
            <AuthContext.Provider value={ contextValue }>
                <MemoryRouter initialEntries={ ['/dc'] }>
                    <DashboardRoutes />
                </MemoryRouter>
            </AuthContext.Provider>
        )

        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h1').text().trim() ).toBe('Dc Screen');


    });

});
