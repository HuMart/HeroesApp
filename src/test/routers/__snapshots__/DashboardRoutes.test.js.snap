// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <DashboardRoutes /> should render correctli in dc screen 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark container-fluid"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="d-flex b"
        height="24"
        src="https://th.bing.com/th/id/OIP.JWMbxW3YwUjNX7Q13lsl6AHaE8?pid=ImgDet&rs=1"
        width="30"
      />
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          <img
            alt=""
            height="24"
            src="https://assets.cdn.moviepilot.de/files/5e6dfbca611668e018c9d231c2bb2caddae173fbf31578a4929c07072f12/fill/1440/691/marvel+logo.jpg"
            width="30"
          />
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
        >
          <img
            alt=""
            height="24"
            src="https://th.bing.com/th/id/R.937784bc187d3ca31114c1461c7c3c20?rik=TCflKvsBsBpSyw&riu=http%3a%2f%2fthebatmanuniverse.net%2fwp-content%2fuploads%2f2018%2f04%2fdc-logo.jpg&ehk=9QBryoGg8QikgHqgU%2fwTzjmk9%2b2ZG8h8KneaVWk6nq8%3d&risl=&pid=ImgRaw&r=0"
            width="30"
          />
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          User logged
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Fly out of here
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Dc Screen
      </h1>
      <hr />
      <div
        className="row rows-cols-1 row-cols-md-3 g-3"
      >
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Batman"
            className="card-img shadow-lg"
            src="/assets/images/dc-batman.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Batman
            </h5>
            <p
              className="card-text fw-bold"
            >
              Bruce Wayne
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-batman"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Superman"
            className="card-img shadow-lg"
            src="/assets/images/dc-superman.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Superman
            </h5>
            <p
              className="card-text fw-bold"
            >
              Kal-El
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-superman"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Flash"
            className="card-img shadow-lg"
            src="/assets/images/dc-flash.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Flash
            </h5>
            <p
              className="card-text fw-bold"
            >
              Jay Garrick
            </p>
            <p
              className="text-bold"
            >
              Jay Garrick, Barry Allen, Wally West, Bart Allen
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-flash"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Green Lantern"
            className="card-img shadow-lg"
            src="/assets/images/dc-green.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Green Lantern
            </h5>
            <p
              className="card-text fw-bold"
            >
              Alan Scott
            </p>
            <p
              className="text-bold"
            >
              Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-green"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Green Arrow"
            className="card-img shadow-lg"
            src="/assets/images/dc-arrow.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Green Arrow
            </h5>
            <p
              className="card-text fw-bold"
            >
              Oliver Queen
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-arrow"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Wonder Woman"
            className="card-img shadow-lg"
            src="/assets/images/dc-wonder.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Wonder Woman
            </h5>
            <p
              className="card-text fw-bold"
            >
              Princess Diana
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-wonder"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Martian Manhunter"
            className="card-img shadow-lg"
            src="/assets/images/dc-martian.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Martian Manhunter
            </h5>
            <p
              className="card-text fw-bold"
            >
              J'onn J'onzz
            </p>
            <p
              className="text-bold"
            >
              Martian Manhunter
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-martian"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Robin/Nightwing"
            className="card-img shadow-lg"
            src="/assets/images/dc-robin.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Robin/Nightwing
            </h5>
            <p
              className="card-text fw-bold"
            >
              Dick Grayson
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-robin"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Blue Beetle"
            className="card-img shadow-lg"
            src="/assets/images/dc-blue.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Blue Beetle
            </h5>
            <p
              className="card-text fw-bold"
            >
              Dan Garret
            </p>
            <p
              className="text-bold"
            >
              Dan Garret, Ted Kord, Jaime Reyes
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-blue"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Black Canary"
            className="card-img shadow-lg"
            src="/assets/images/dc-black.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Black Canary
            </h5>
            <p
              className="card-text fw-bold"
            >
              Dinah Drake
            </p>
            <p
              className="text-bold"
            >
              Dinah Drake, Dinah Lance
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/dc-black"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;

exports[`test in <DashboardRoutes /> should render correctli in marvel screen 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark container-fluid"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      <img
        alt=""
        className="d-flex b"
        height="24"
        src="https://th.bing.com/th/id/OIP.JWMbxW3YwUjNX7Q13lsl6AHaE8?pid=ImgDet&rs=1"
        width="30"
      />
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          className="nav-item nav-link "
          href="/marvel"
          onClick={[Function]}
        >
          <img
            alt=""
            height="24"
            src="https://assets.cdn.moviepilot.de/files/5e6dfbca611668e018c9d231c2bb2caddae173fbf31578a4929c07072f12/fill/1440/691/marvel+logo.jpg"
            width="30"
          />
        </a>
        <a
          className="nav-item nav-link "
          href="/dc"
          onClick={[Function]}
        >
          <img
            alt=""
            height="24"
            src="https://th.bing.com/th/id/R.937784bc187d3ca31114c1461c7c3c20?rik=TCflKvsBsBpSyw&riu=http%3a%2f%2fthebatmanuniverse.net%2fwp-content%2fuploads%2f2018%2f04%2fdc-logo.jpg&ehk=9QBryoGg8QikgHqgU%2fwTzjmk9%2b2ZG8h8KneaVWk6nq8%3d&risl=&pid=ImgRaw&r=0"
            width="30"
          />
        </a>
        <a
          className="nav-item nav-link "
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          User logged
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Fly out of here
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="row rows-cols-1 row-cols-md-3 g-3"
      >
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Spider Man"
            className="card-img shadow-lg"
            src="/assets/images/marvel-spider.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Spider Man
            </h5>
            <p
              className="card-text fw-bold"
            >
              Peter Parker
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-spider"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Captain America"
            className="card-img shadow-lg"
            src="/assets/images/marvel-captain.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Captain America
            </h5>
            <p
              className="card-text fw-bold"
            >
              Steve Rogers
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-captain"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Iron Man"
            className="card-img shadow-lg"
            src="/assets/images/marvel-iron.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Iron Man
            </h5>
            <p
              className="card-text fw-bold"
            >
              Tony Stark
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-iron"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Thor"
            className="card-img shadow-lg"
            src="/assets/images/marvel-thor.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Thor
            </h5>
            <p
              className="card-text fw-bold"
            >
              Thor Odinson
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-thor"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Hulk"
            className="card-img shadow-lg"
            src="/assets/images/marvel-hulk.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Hulk
            </h5>
            <p
              className="card-text fw-bold"
            >
              Bruce Banner
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-hulk"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Wolverine"
            className="card-img shadow-lg"
            src="/assets/images/marvel-wolverine.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Wolverine
            </h5>
            <p
              className="card-text fw-bold"
            >
              James Howlett
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-wolverine"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Daredevil"
            className="card-img shadow-lg"
            src="/assets/images/marvel-daredevil.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Daredevil
            </h5>
            <p
              className="card-text fw-bold"
            >
              Matthew Michael Murdock
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-daredevil"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Hawkeye"
            className="card-img shadow-lg"
            src="/assets/images/marvel-hawkeye.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Hawkeye
            </h5>
            <p
              className="card-text fw-bold"
            >
              Clinton Francis Barton
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-hawkeye"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Cyclops"
            className="card-img shadow-lg"
            src="/assets/images/marvel-cyclops.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Cyclops
            </h5>
            <p
              className="card-text fw-bold"
            >
              Scott Summers
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-cyclops"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
        <div
          className="card  border-0 bg-transparent text-black animate__animated animate__fadeIn"
        >
          <img
            alt="Silver Surfer"
            className="card-img shadow-lg"
            src="/assets/images/marvel-silver.jpg"
          />
          <div
            className=" card-img-overlay "
          >
            <h5
              className="card-title "
            >
              Silver Surfer
            </h5>
            <p
              className="card-text fw-bold"
            >
              Norrin Radd
            </p>
            <div
              className="col-md-8"
            >
              <div
                className="card-body"
              >
                <a
                  className="text-black fw-bold"
                  href="/hero/marvel-silver"
                  onClick={[Function]}
                >
                  More...
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
